src/scripts.js

```diff
@@ -37,32 +41,40 @@ function createRecipeCards(recipes) {
         recipeSection.innerHTML += `
         <article class="recipe-card" data-parent="${recipe.id}">
             <img class="recipe-img" src="${recipe.image}" data-parent="${recipe.id}" alt="picture of ${recipe.name}">
-            <img class="star-icon hidden" id="star-icon" data-parent="${recipe.id}" src="./images/star-icon.png" alt="This recipe is in my recipes!">
+            <img class="heart-icon hidden" id="heartIcon" data-parent="${recipe.id}" src="./images/heart.png" alt="This recipe is in my recipes!">
             <h3 style="font-size: ${size}rem" data-parent="${recipe.id}">${recipe.name}</h3>
         </article>`;
+        let heart = document.getElementById('heartIcon')
+        if(recipe.saved){
+            toggleHidden(heart)
+        }
     });
 }
 
 function createRecipeModal(event) {
   toggleHidden(modalSection);
   let recipeID = +(event.target.dataset.parent);
-  let clickedRecipe = recipeRepo.recipes.find(recipe => recipe.id === recipeID);
+  modalRecipe = recipeRepo.recipes.find(recipe => recipe.id === recipeID);
+  let buttonText
+  modalRecipe.saved ? buttonText = "Remove from Saved Recipes" : buttonText = "Add to Saved Recipes"
   modalSection.innerHTML = `
   <div class="recipe-popup">
-      <h2>${clickedRecipe.name}</h2>
+      <h2>${modalRecipe.name}</h2>
       <div class="image-ingredients">
-      <img class="recipe-img" src="${clickedRecipe.image}" alt="${clickedRecipe.name} image">
+      <img class="recipe-img" src="${modalRecipe.image}" alt="${modalRecipe.name} image">
       <ul class="ingredient-list">
           <h3>Ingredients:</h3>
-          ${createList(clickedRecipe.listIngredients(ingredientsData))}
+          ${createList(modalRecipe.listIngredients(ingredientsData))}
       </ul>
       </div>
       <ol class="direction-list">
       <h3>Directions:</h3>
-      ${createList(clickedRecipe.getInstructions())}
+      ${createList(modalRecipe.getInstructions())}
       </ol>
-      <h4>TOTAL COST $${+(clickedRecipe.listCost(ingredientsData))}</h4>
+      <h4>TOTAL COST $${+(modalRecipe.listCost(ingredientsData))}</h4>
+      <button class="save-button" id="saveBtn">${buttonText}</button>
   </div>`;
+  document.getElementById('saveBtn').addEventListener('click', toggleSaveRecipe);
```

Comment:
May want to revisit this later. Even though we're regenerating the HTML for this section with each popup modal, is it possible that we would still have eventListeners out there lingering? If so, we may want to change the element we're listening on to the modal (or add a conditional to check if the event is happening on the button?) and then change our handler to account for that.